# PCR Primer Diagnostic Kits Design for Covid-19 (Alpha- Delta- Omicron) in Africa 🧬🌍

Welcome to our Bioinformatics project dedicated to designing PCR primer diagnostic kits for COVID-19 variants (Alpha, Delta, Omicron) specifically in Africa.

## Project Overview 🚀

This venture marks my first journey into the realm of Bioinformatics, and I couldn't be more thrilled to share it with you. Our primary objective was to craft a versatile script capable of generating primers for any disease, provided we have the relevant sequences. However, this time around, our focus was on the ever-evolving landscape of COVID-19, with a special emphasis on its variants within the African context.

## Project Highlights 🌟

- **Innovation in Action:** Witness the application of cutting-edge bioinformatics techniques in the development of diagnostic tools for a critical global health challenge.

- **Script Power:** Explore the inner workings of our general-purpose script designed to generate primers, offering a glimpse into the magic behind disease-specific kit creation.

## Enjoy the Journey 🌐

Take a tour through our project, and I hope you find the exploration of PCR primer design both insightful and enjoyable. Your feedback is not just welcome; it's eagerly anticipated! 😃

### Data Collection

This project addresses the evolving landscape of COVID-19, a zoonotic disease first discovered in Wuhan, China, in December 2019. The World Health Organization (WHO) declared it a pandemic on March 11, 2020. As part of our exploration, we delve into the characteristics of key variants that have shaped the trajectory of the pandemic.

### Variants

#### Alpha Variant (B.1.1.7)
The Alpha variant, identified by the lineage B.1.1.7, stands as the inaugural variant of SARS-CoV-2, providing critical insights into the virus's early evolution.

#### Delta Variant (B.1.617.2)
Characterized by the lineage B.1.617.2, the Delta variant has proven to be more transmissible than its predecessor, the Alpha variant, presenting heightened challenges in containment efforts.

#### Omicron Variant (B.1.1.529)
The Omicron variant, encompassing lineages such as BA.1, BA.1.1, BA.2, and BA.3 within the lineage B.1.1.529, marks a notable evolution of the virus, introducing new complexities in our ongoing battle against the pandemic.

![](/Pictures/Covid%20Variants.jpeg)
 PX Media / Adobe Stock

## Pango Lineages Exploration

Our research initiative involves a meticulous exploration of COVID-19 Pango lineages. Leveraging the NCBI virus database, we focus on crucial lineages—BA.1, B.1.617.2, and B.1.1.7—to deepen our understanding of the Omicron, Delta, and Alpha variants. This exploration contributes essential insights to the broader knowledge of the virus, aiding in the development of effective strategies for mitigation and prevention.

### Results

#### Alpa, Delta,and Omicron primers 
For primer design, we harnessed Primer-BLAST, leveraging the RefSeq RNA database for template identification. Default parameters sufficed for primer range definition due to template lengths (less than 50,000 bp).
The designed primers, essential to our methodology, are presented below:



#### Universal primers 
Using only 70 sequences from each variant we believe our produced primer can detect any Alpha, Beta, or Omicron strain. Upon validation, our primer passed all the PCR primer stats tests and UCSC in-Silico PCR.



## Special Thanks 🙌

A heartfelt shoutout to my incredible team members whose dedication and hard work made this project possible:

- Ahmed Elghamry
- Nour Bahaa
- Ahmed Hussein
- Yasmine Hosny
- Kareem Muhammad

Feel free to dive into the code, explore our process, and share your thoughts. Together, let's advance the frontiers of Bioinformatics! 🧪🔬
